llms/googleai/googleai.go

```diff
@@ -50,6 +50,7 @@ func (g *GoogleAI) GenerateContent(ctx context.Context, messages []llms.MessageC
 	for _, opt := range options {
 		opt(&opts)
 	}
+	g.setCallOptionsDefaults(&opts)
```

Comment:
The defaults were just set a few lines up; this doesn't seem right - we should be setting defaults in one place.

Also I don't understand the logic here -- for example, what if the user requested Temperature = 0 explicitly? That's a valid request, why do we override it with a default now?

Reply:
> Temperature = 0
I later realized this issue: the current code cannot distinguish whether a zero value means it's unset or intentionally set to zero. Currently, only `topP` and `Temperature` can be set to zero.

The correct way I guess might be to modify `getLLMCallOptions` so that it only changes the passed parameters to `lms.CallOptions`? But in that case, it will affect all LLM... Maybe more testing is needed...

Reply:
Please revert this PR until we figure out the right way to address this, and create an issue describing the problem where it can be discussed.

Reply:
> create an issue 

#626 

> revert this PR

IMO, the pr that fixes this problem **completely** can revert this commit before submitting their own fix, the current approach has small bug(Can't set `topP` and `Temperature` to 0 on chain calls), but at least it ensures that google ai is working and can return results, without the current pr, chain call without full ChainCallOption will only return 404...

```
TestLLMChainWithGoogleAI
Entering LLM with messages:
Role: human
Text: What is the capital of France
2024/02/22 10:15:23 googleapi: Error 404:
```